lib/wrappers/tokens/erc20.ts

```diff
@@ -0,0 +1,62 @@
+import {
+    Contract
+} from "ethers";
+
+import {
+    Address,
+    AddressOrAddressable,
+    SupportedProvider
+} from "../../types";
+import BaseWrapper from "../../base-wrapper";
+import { ERC20 } from "../../../typechain-types";
+import { abi as ERC20ABI } from "../../../artifacts/@openzeppelin/contracts/token/ERC20/ERC20.sol/ERC20.json"
+
+export class ERC20Wrapper extends BaseWrapper {
+    constructor(
+        address: AddressOrAddressable, 
+        provider: SupportedProvider
+    ) {
+        const contract = new Contract(address, ERC20ABI, provider)
+        super(contract, provider)
+    }
+
+    async totalSupply(): Promise<bigint> {
+        return this.call.totalSupply()
+    }
+
+    async balanceOf(account: Address): Promise<bigint> {
+        return this.call.balanceOf(account)
+    }
+
+    async transfer(to: Address, value: string) {
```

Comment:
You should wait a number of confirmations before returning. I think there is a helper for that in the base class.

Reply:
Fixed in ab9fd61